packages/sdk/test/evm/new-publish-deploy.test.ts

```diff
@@ -0,0 +1,270 @@
+import { SignerWithAddress } from "@nomiclabs/hardhat-ethers/signers";
+import { jsonProvider, sdk, signers, thirdwebFactory } from "./before-setup";
+import { expect, assert } from "chai";
+import { ThirdwebSDK, fetchAndCacheDeployMetadata } from "../../src/evm";
+import { BigNumber } from "ethers";
+import { mockUploadWithCID } from "./mock/MockStorage";
+
+describe("New Publish Flow", async () => {
+  let adminWallet: SignerWithAddress,
+    samWallet: SignerWithAddress,
+    bobWallet: SignerWithAddress,
+    w4: SignerWithAddress;
+
+  let realSDK: ThirdwebSDK;
+
+  beforeEach(async () => {
+    await jsonProvider.send("hardhat_reset", []);
+    [adminWallet, samWallet, bobWallet, , , , , w4] = signers;
+    realSDK = new ThirdwebSDK(adminWallet, {
+      gatewayUrls: ["https://gateway.ipfscdn.io/ipfs/"],
+    });
+  });
+
+  describe("Deploy newly published contract", async () => {
+    let publishUriWithDirectDeploy: string;
+    let publishUriWithCustomFactoryDeploy: string;
+    let customFactoryPublishUri: string;
+
+    before("", async () => {
+      // https://thirdweb.com/thirdweb.eth/MyContract - v1.0.1
+      publishUriWithDirectDeploy =
+        "ipfs://QmUCEvfDQxEmWNLrcun8ZV8pUU4x9oxkFavJjC133tJFVN/";
+
+      // Split Wallet
+      publishUriWithCustomFactoryDeploy =
+        "ipfs://Qmc5cQotuEnCo71ZztJk69wc8kuKbGmuCARL2FyAojoi9r/";
+
+      // Split Main
+      customFactoryPublishUri =
+        "ipfs://QmZZUsE2hQbKc4pHPC3qjfXYrMvg8mceDqHxoQhgMudwau/";
+    });
+    it("should deploy regular contract", async () => {
+      const input = 27;
+      const contractAddress = await realSDK.deployer.deployContractFromUri(
+        publishUriWithDirectDeploy,
+        [input],
+      );
+      const contract = await realSDK.getContract(contractAddress);
+
+      const number: BigNumber = await contract.call("number");
+      const deployer: string = await contract.call("deployer");
+
+      expect(number.toNumber()).to.equal(input);
+      expect(deployer).to.equal(adminWallet.address);
+    });
+
+    it("should deploy with custom factory", async () => {
+      // Re-upload with mock storage, enable all networks
+      const { compilerMetadata, extendedMetadata } =
+        await fetchAndCacheDeployMetadata(
+          customFactoryPublishUri,
+          realSDK.storage,
```

Comment:
same here, you're downloading this data just to put it in mock storage, can't we just hardcode the data in our tests assets?